### PR TITLE
day two groups and transactions

### PR DESCRIPTION
**Pull request by paulshaun-kdtk**
# Changes 

- implemented transactions and rollbacks
- queried table based on groups
- modified animals table to include a pk on the id field

## results screenshots:

Screenshot (261)
Screenshot (262)
Screenshot (263)
Screenshot (260)

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 CREATE TABLE animals (
-    id INTEGER,
+    id SERIAL PRIMARY KEY,
     name VARCHAR(255),
     date_of_birth DATE,
     escape_attempts INTEGER,
     neutered BOOLEAN,
-    weight_kg DECIMAL(8, 2)
+    weight_kg DECIMAL(8, 2),
+    species VARCHAR(255)
```
Comment (joskalenda) [inline schema.sql]: - [ ] Well, you are asked to add a column to a table, and you do but in an unprofessional way, we are expecting you to use appropriate to add something to a given table, try reading on the `ALTER` command and update your project with the new info you will gain

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @paulshaun-kdtk , |
|Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! |
|  |

## Highlights

- [ ].. ✅ Good code
- [ ].. ✅ Clean commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Abenezer-Tilahun) [reply schema.sql]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @paulshaun-kdtk 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow. 
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

- [ ] Kindly make sure that for each project activity Use descriptive commit messages - each commit should be small enough to describe it in one sentence. The commit message should be written in the imperative tense and capitalized and Add descriptive README file to your project with each section of the README file address the current project activity. 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**